src/main/java/jenkins/security/plugins/ldap/FromUserRecordLDAPGroupMembershipStrategy.java

```diff
@@ -102,37 +101,37 @@ public String getAttributeName() {
                     String role = ga.getAuthority();
 
                     // backward compatible name mangling
-                    if (authoritiesPopulatorImpl.isConvertToUpperCase()) {
+                    if (authoritiesPopulatorImpl._isConvertToUpperCase()) {
                         role = role.toUpperCase();
                     }
-                    GrantedAuthorityImpl extraAuthority = new GrantedAuthorityImpl(
-                            authoritiesPopulatorImpl.getRolePrefix() + role);
+                    GrantedAuthority extraAuthority = new SimpleGrantedAuthority(
+                            authoritiesPopulatorImpl._getRolePrefix() + role);
                     result.add(extraAuthority);
                 }
             }
-            result.addAll(authoritiesPopulatorImpl.getAdditionalRoles(ldapUser));
+            result.addAll(authoritiesPopulatorImpl.getAdditionalRoles(userData, /* TODO currently ignored anyway, but where does username come from? */null));
```

Comment:
Maybe needs to be passed in as an additional argument or something?

Reply:
isn't it the second argument to the method you are in?